src/viam/sdk/components/private/board_client.cpp

```diff
@@ -137,6 +139,33 @@ Board::digital_value BoardClient::read_digital_interrupt(const std::string& digi
     return response.value();
 }
 
+void BoardClient::stream_ticks(const std::vector<std::string> digital_interrupt_names,
+                               std::shared_ptr<std::queue<tick>> ticks,
+                               const AttributeMap& extra) {
+    viam::component::board::v1::StreamTicksRequest request;
+    viam::component::board::v1::StreamTicksResponse response;
+    ClientContext ctx;
+
+    request.set_name(this->name());
+
+    for (unsigned int i = 0; i < digital_interrupt_names.size(); i++) {
+        request.add_pin_names(digital_interrupt_names[i]);
+    }
+    *request.mutable_extra() = map_to_struct(extra);
+
+    std::unique_ptr<
+        ::grpc::ClientReaderInterface<::viam::component::board::v1::StreamTicksResponse>>
+        reader = stub_->StreamTicks(ctx, request);
+
+    while (reader->Read(&response)) {
+        Board::tick tick;
+        tick.pin_name = response.pin_name();
+        tick.high = response.high();
+        tick.time = response.time();
+        ticks->push(tick);
```

Comment:
This can probably be simplified to:
```
while (...) {
    ticks->push({std::move(response.pin_name()), std::move(response.high()), std::move(response.time()};
}
```